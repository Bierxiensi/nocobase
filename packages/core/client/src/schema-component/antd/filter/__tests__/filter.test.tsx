import { AntdSchemaComponentProvider, Input, SchemaComponent, SchemaComponentProvider } from '@nocobase/client';
import React from 'react';
import { Filter } from '../Filter';
import { render, screen, fireEvent, act } from '@testing-library/react';

const schema: any = {
  type: 'void',
  properties: {
    demo: {
      name: 'filter',
      type: 'object',
      enum: [
        {
          name: 'name',
          title: 'Name',
          operators: [
            { label: 'eq', value: '$eq' },
            { label: 'ne', value: '$ne' },
          ],
          schema: {
            type: 'string',
            title: 'Name',
            'x-component': 'Input',
          },
        },
        {
          name: 'age',
          title: 'Age',
          operators: [
            { label: 'in', value: '$in' },
            { label: 'not', value: '$not' },
          ],
          schema: {
            type: 'string',
            title: 'Age',
            'x-component': 'InputNumber',
          },
        },
        {
          name: 'tags',
          title: 'Tags',
          schema: {
            title: 'Tags',
          },
          children: [
            {
              name: 'slug',
              title: 'Slug',
              operators: [
                { label: 'in', value: '$in' },
                { label: 'not', value: '$not' },
              ],
              schema: {
                title: 'Slug',
                type: 'string',
                'x-component': 'Input',
              },
            },
            {
              name: 'title',
              title: 'Title',
              operators: [
                { label: 'eq', value: '$eq' },
                { label: 'ne', value: '$ne' },
              ],
              schema: {
                title: 'Title',
                type: 'string',
                'x-component': 'Input',
              },
            },
          ],
        },
      ],
      default: {
        $or: [
          {
            name: {
              $ne: null,
            },
          },
          {
            'tags.title': {
              $eq: 'aaa',
            },
          },
        ],
      },
      'x-component': 'Filter',
      'x-component-props': {},
    },
  },
};

const App = () => {
  return (
    <SchemaComponentProvider>
      <AntdSchemaComponentProvider>
        <SchemaComponent components={{ Input, Filter }} schema={schema} />
      </AntdSchemaComponentProvider>
    </SchemaComponentProvider>
  );
};

describe('Filter', () => {
  it('filter add condition', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
    const addBtn = screen.getByText('Add condition');
    fireEvent.click(addBtn);
    const selectField = container.getElementsByClassName('ant-select-selection-placeholder')[0];
    expect(selectField.textContent).toBe('Select Field');
  });
});
