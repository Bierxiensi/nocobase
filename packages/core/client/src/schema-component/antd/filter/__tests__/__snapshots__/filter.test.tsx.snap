// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter filter 1`] = `
<div>
  <div>
    <div
      style="position: relative; margin-bottom: 8px;"
    >
      <div
        style="margin-bottom: 8px;"
      >
        Meet 
        <div
          class="ant-select ant-select-single ant-select-show-arrow"
          style="width: auto;"
        >
          <div
            class="ant-select-selector"
          >
            <span
              class="ant-select-selection-search"
            >
              <input
                aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                aria-autocomplete="list"
                aria-controls="rc_select_TEST_OR_SSR_list"
                aria-haspopup="listbox"
                aria-owns="rc_select_TEST_OR_SSR_list"
                autocomplete="off"
                class="ant-select-selection-search-input"
                id="rc_select_TEST_OR_SSR"
                readonly=""
                role="combobox"
                style="opacity: 0;"
                type="search"
                unselectable="on"
                value=""
              />
            </span>
            <span
              class="ant-select-selection-item"
            >
              Any
            </span>
          </div>
          <span
            aria-hidden="true"
            class="ant-select-arrow"
            style="user-select: none;"
            unselectable="on"
          >
            <span
              aria-label="down"
              class="anticon anticon-down ant-select-suffix"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </span>
        </div>
         conditions in the group
      </div>
      <div>
        <div>
          <div
            style="margin-bottom: 8px;"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <div
                  class="ant-select ant-cascader css-obaqht ant-select-single ant-select-allow-clear ant-select-show-arrow"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title="Name"
                    >
                      Name
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    aria-hidden="true"
                    class="ant-select-clear"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="close-circle"
                      class="anticon anticon-close-circle"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="close-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <div
                  class="ant-select ant-select-single ant-select-show-arrow"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title="ne"
                    >
                      ne
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <span
                  class="ant-input-affix-wrapper"
                  style="min-width: 150px;"
                >
                  <input
                    class="ant-input"
                    type="text"
                    value=""
                  />
                  <span
                    class="ant-input-suffix"
                  >
                    <span />
                  </span>
                </span>
              </div>
              <div
                class="ant-space-item"
              >
                <a>
                  <span
                    aria-label="close-circle"
                    class="anticon anticon-close-circle"
                    role="img"
                    style="color: rgb(191, 191, 191);"
                    tabindex="-1"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M685.4 354.8c0-4.4-3.6-8-8-8l-66 .3L512 465.6l-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155L340.5 670a8.32 8.32 0 00-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3L512 564.4l99.3 118.4 66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.5 515l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
                      />
                      <path
                        d="M512 65C264.6 65 64 265.6 64 513s200.6 448 448 448 448-200.6 448-448S759.4 65 512 65zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                      />
                    </svg>
                  </span>
                </a>
              </div>
            </div>
          </div>
          <div
            style="margin-bottom: 8px;"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <div
                  class="ant-select ant-cascader css-obaqht ant-select-single ant-select-allow-clear ant-select-show-arrow"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title="Tags / Title"
                    >
                      Tags / Title
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    aria-hidden="true"
                    class="ant-select-clear"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="close-circle"
                      class="anticon anticon-close-circle"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="close-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <div
                  class="ant-select ant-select-single ant-select-show-arrow"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title="eq"
                    >
                      eq
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <span
                  class="ant-input-affix-wrapper"
                  style="min-width: 150px;"
                >
                  <input
                    class="ant-input"
                    type="text"
                    value=""
                  />
                  <span
                    class="ant-input-suffix"
                  >
                    <span />
                  </span>
                </span>
              </div>
              <div
                class="ant-space-item"
              >
                <a>
                  <span
                    aria-label="close-circle"
                    class="anticon anticon-close-circle"
                    role="img"
                    style="color: rgb(191, 191, 191);"
                    tabindex="-1"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M685.4 354.8c0-4.4-3.6-8-8-8l-66 .3L512 465.6l-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155L340.5 670a8.32 8.32 0 00-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3L512 564.4l99.3 118.4 66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.5 515l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
                      />
                      <path
                        d="M512 65C264.6 65 64 265.6 64 513s200.6 448 448 448 448-200.6 448-448S759.4 65 512 65zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                      />
                    </svg>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-space ant-space-horizontal ant-space-align-center"
        style="margin-top: 8px; margin-bottom: 8px;"
      >
        <div
          class="ant-space-item"
          style="margin-right: 16px;"
        >
          <a>
            Add condition
          </a>
        </div>
        <div
          class="ant-space-item"
        >
          <a>
            Add condition group
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Filter filter add condition 1`] = `
<div>
  <div>
    <div
      style="position: relative; margin-bottom: 8px;"
    >
      <div
        style="margin-bottom: 8px;"
      >
        Meet 
        <div
          class="ant-select ant-select-single ant-select-show-arrow"
          style="width: auto;"
        >
          <div
            class="ant-select-selector"
          >
            <span
              class="ant-select-selection-search"
            >
              <input
                aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                aria-autocomplete="list"
                aria-controls="rc_select_TEST_OR_SSR_list"
                aria-haspopup="listbox"
                aria-owns="rc_select_TEST_OR_SSR_list"
                autocomplete="off"
                class="ant-select-selection-search-input"
                id="rc_select_TEST_OR_SSR"
                readonly=""
                role="combobox"
                style="opacity: 0;"
                type="search"
                unselectable="on"
                value=""
              />
            </span>
            <span
              class="ant-select-selection-item"
            >
              Any
            </span>
          </div>
          <span
            aria-hidden="true"
            class="ant-select-arrow"
            style="user-select: none;"
            unselectable="on"
          >
            <span
              aria-label="down"
              class="anticon anticon-down ant-select-suffix"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </span>
        </div>
         conditions in the group
      </div>
      <div>
        <div>
          <div
            style="margin-bottom: 8px;"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <div
                  class="ant-select ant-cascader css-obaqht ant-select-single ant-select-allow-clear ant-select-show-arrow"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title="Name"
                    >
                      Name
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    aria-hidden="true"
                    class="ant-select-clear"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="close-circle"
                      class="anticon anticon-close-circle"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="close-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <div
                  class="ant-select ant-select-single ant-select-show-arrow"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title="ne"
                    >
                      ne
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <span
                  class="ant-input-affix-wrapper"
                  style="min-width: 150px;"
                >
                  <input
                    class="ant-input"
                    type="text"
                    value=""
                  />
                  <span
                    class="ant-input-suffix"
                  >
                    <span />
                  </span>
                </span>
              </div>
              <div
                class="ant-space-item"
              >
                <a>
                  <span
                    aria-label="close-circle"
                    class="anticon anticon-close-circle"
                    role="img"
                    style="color: rgb(191, 191, 191);"
                    tabindex="-1"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M685.4 354.8c0-4.4-3.6-8-8-8l-66 .3L512 465.6l-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155L340.5 670a8.32 8.32 0 00-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3L512 564.4l99.3 118.4 66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.5 515l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
                      />
                      <path
                        d="M512 65C264.6 65 64 265.6 64 513s200.6 448 448 448 448-200.6 448-448S759.4 65 512 65zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                      />
                    </svg>
                  </span>
                </a>
              </div>
            </div>
          </div>
          <div
            style="margin-bottom: 8px;"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <div
                  class="ant-select ant-cascader css-obaqht ant-select-single ant-select-allow-clear ant-select-show-arrow"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title="Tags / Title"
                    >
                      Tags / Title
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    aria-hidden="true"
                    class="ant-select-clear"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="close-circle"
                      class="anticon anticon-close-circle"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="close-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <div
                  class="ant-select ant-select-single ant-select-show-arrow"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title="eq"
                    >
                      eq
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <span
                  class="ant-input-affix-wrapper"
                  style="min-width: 150px;"
                >
                  <input
                    class="ant-input"
                    type="text"
                    value=""
                  />
                  <span
                    class="ant-input-suffix"
                  >
                    <span />
                  </span>
                </span>
              </div>
              <div
                class="ant-space-item"
              >
                <a>
                  <span
                    aria-label="close-circle"
                    class="anticon anticon-close-circle"
                    role="img"
                    style="color: rgb(191, 191, 191);"
                    tabindex="-1"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M685.4 354.8c0-4.4-3.6-8-8-8l-66 .3L512 465.6l-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155L340.5 670a8.32 8.32 0 00-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3L512 564.4l99.3 118.4 66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.5 515l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
                      />
                      <path
                        d="M512 65C264.6 65 64 265.6 64 513s200.6 448 448 448 448-200.6 448-448S759.4 65 512 65zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                      />
                    </svg>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-space ant-space-horizontal ant-space-align-center"
        style="margin-top: 8px; margin-bottom: 8px;"
      >
        <div
          class="ant-space-item"
          style="margin-right: 16px;"
        >
          <a>
            Add condition
          </a>
        </div>
        <div
          class="ant-space-item"
        >
          <a>
            Add condition group
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
